src/systems/filecoin_mining/storage_mining/storage_miner_actor_code.go

```diff
@@ -580,19 +580,21 @@ func (a *StorageMinerActorCode_I) _verifySeal(rt Runtime, onChainInfo sector.OnC
 // TODO(optimization): PreCommitSector could contain a list of deals that are not published yet.
 func (a *StorageMinerActorCode_I) PreCommitSector(rt Runtime, info sector.SectorPreCommitInfo) InvocOutput {
 	TODO() // TODO: validate caller
-
 	// can be called regardless of Challenged status
 
-	// TODO: might take collateral in case no ProveCommit follows within sometime
-	// TODO: collateral also penalizes repeated precommit to get randomness that one likes
 	// TODO: might be a good place for Treasury
 
 	h, st := a.State(rt)
 
+	msgValue := rt.ValueReceived()
```

Comment:
I think it's worth documenting somewhere up the top that the balance of a storage miner actor should equal exactly the sum of precommit deposits not yet either returned or burnt.

Reply:
ack, will do